src/coreComponents/constitutive/relativePermeability/TableRelativePermeabilityHysteresis.cpp

```diff
@@ -35,113 +35,125 @@ TableRelativePermeabilityHysteresis::TableRelativePermeabilityHysteresis( std::s
+    " to specify the table names" );
 
   // imbibition table names
 
   registerWrapper( viewKeyStruct::imbibitionWettingRelPermTableNameString(), &m_imbibitionWettingRelPermTableName ).
+    setInputFlag(
+    InputFlags::OPTIONAL ).
+    setApplyDefaultValue(
+    "" ).
+    setDescription(
+    "Imbibition relative permeability table name for the wetting phase.\n"
+    "To neglect hysteresis on this phase, just use the same table name for the drainage and imbibition curves" );
+
+  registerWrapper( viewKeyStruct::imbibitionNonWettingRelPermTableNameString(),
+                   &m_imbibitionNonWettingRelPermTableName ).
     setInputFlag( InputFlags::OPTIONAL ).
     setApplyDefaultValue( "" ).
-    setDescription( "Imbibition relative permeability table name for the wetting phase.\n"
-                    "To neglect hysteresis on this phase, just use the same table name for the drainage and imbibition curves" );
-
-  registerWrapper( viewKeyStruct::imbibitionNonWettingRelPermTableNameString(), &m_imbibitionNonWettingRelPermTableName ).
-    setInputFlag( InputFlags::OPTIONAL ).
-    setApplyDefaultValue( "" ).
-    setDescription( "Imbibition relative permeability table name for the non-wetting phase.\n"
-                    "To neglect hysteresis on this phase, just use the same table name for the drainage and imbibition curves" );
+    setDescription(
+    "Imbibition relative permeability table name for the non-wetting phase.\n"
+    "To neglect hysteresis on this phase, just use the same table name for the drainage and imbibition curves" );
 
   // hysteresis input parameters
-
-  registerWrapper( viewKeyStruct::jerauldParameterAString(), &m_jerauldParam_a ).
-    setInputFlag( InputFlags::OPTIONAL ).
-    setApplyDefaultValue( 0.1 ).
-    setDescription( "First parameter (modification parameter) introduced by Jerauld in the Land trapping model (see RTD documentation)." );
-
-  registerWrapper( viewKeyStruct::jerauldParameterBString(), &m_jerauldParam_b ).
-    setInputFlag( InputFlags::OPTIONAL ).
-    setApplyDefaultValue( 0.0 ).
-    setDescription( "Second parameter introduced by Jerauld in the Land trapping model (see RTD documentation)." );
-
-  registerWrapper( viewKeyStruct::killoughCurvatureParameterString(), &m_killoughCurvatureParam ).
-    setInputFlag( InputFlags::OPTIONAL ).
-    setApplyDefaultValue( 1.0 ).
-    setDescription( "Curvature parameter introduced by Killough for wetting-phase hysteresis (see RTD documentation)." );
-
-  // internal class data
-
-  registerWrapper( viewKeyStruct::drainagePhaseMinVolumeFractionString(), &m_drainagePhaseMinVolFraction ).
-    setInputFlag( InputFlags::FALSE ). // will be deduced from tables
+  registerWrapper( viewKeyStruct::phaseHasHysteresisString(), &m_phaseHasHysteresis ).
+    setInputFlag( InputFlags::FALSE )
+    .                                                                                  // will be deduced from tables
```

Comment:
Maybe re-organize slightly the registrations to have 1) scalars, 2) structures, 3) registerFields

Reply:
Done